pyemma/coordinates/transform/transformer.py

```diff
@@ -35,15 +35,89 @@
 __all__ = ['Transformer']
 __author__ = 'noe, marscher'
 
+class TransformerIteratorContext(object):
+
+    def __init__(self, stride=1, lag=0):
+        self._lag = lag
+        self.__init_stride(stride)
+
+    def __init_stride(self, stride):
+        self._stride = stride
+        if isinstance(stride, np.ndarray):
+            keys = stride[:, 0]
+            self._trajectory_keys, self._trajectory_lengths = np.unique(keys, return_counts=True)
+        else:
+            self._trajectory_keys = None
+        self._uniform_stride = TransformerIteratorContext.is_uniform_stride(stride)
+        if not self.uniform_stride and not self.is_stride_sorted():
+            raise ValueError("Currently only sorted arrays allowed for random access")
+
+    def ra_indices_for_traj(self, traj):
+        """
+        Gives the indices for a trajectory file (without changing the order within the trajectory itself).
+        :param traj:
+        :return:
+        """
+        assert not self.uniform_stride, "requested random access indices, but is in uniform stride mode"
+        return self._stride[self._stride[:, 0] == traj][:, 1] if traj in self.traj_keys else np.array([])
```

Comment:
Actually you should use the new stride semantics in eg. save_traj to
obtain the frames efficiently to avoid code duplication or
re-implementing the same thing, because I think @clonker has done this
the most (io) efficient way possible.